internal/interval/validate_test.go

```diff
@@ -0,0 +1,47 @@
+package interval
+
+import (
+	"errors"
+	"testing"
+)
+
+func TestValidateForDefrag(t *testing.T) {
+	t.Parallel()
+
+	tests := []struct {
+		name   string
+		input  string
+		output string
+		err    error
+	}{
+		{"valid value hourly", "hourly", "@hourly", nil},
+		{"valid value daily", "daily", "@daily", nil},
+		{"valid value weekly", "weekly", "@weekly", nil},
+		{"empty", "", "", ErrInvalidDefragInterval},
+		{"invalid value", "*/1 * * * *", "", ErrInvalidDefragInterval},
+	}
+
+	for _, tc := range tests {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			out, err := ValidateForDefrag(tc.input)
+
+			if tc.err != nil && err == nil {
```

Comment:
I would recommend using testify's assertions here
https://github.com/stretchr/testify

`assert.Equal(t, tc.err, err)`

Reply:
The v1.6.1 version did not support wrapped errors, I updated the library to v.1.7.2 which has methods for checking wrapped errors. That said, the methods do not handle both nil and error cases in the same method so I added a small helper method https://github.com/storageos/etcd-cluster-operator/pull/37/files#diff-f4cb5f477ae509d72e01582438b6e1b118c6695ac2287f765d3e0f10c19460c3R31